services/audit-service/src/main/java/datawave/microservice/audit/replay/ReplayController.java

```diff
@@ -0,0 +1,726 @@
+package datawave.microservice.audit.replay;
+
+import datawave.microservice.audit.AuditController;
+import datawave.microservice.audit.config.AuditProperties;
+import datawave.microservice.audit.replay.config.ReplayProperties;
+import datawave.microservice.audit.replay.remote.Request;
+import datawave.microservice.audit.replay.runner.ReplayTask;
+import datawave.microservice.audit.replay.runner.RunningReplay;
+import datawave.microservice.audit.replay.status.Status;
+import datawave.microservice.audit.replay.status.StatusCache;
+import datawave.webservice.common.audit.AuditParameters;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.bus.BusProperties;
+import org.springframework.cloud.bus.event.AuditReplayRemoteRequestEvent;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.task.TaskRejectedException;
+import org.springframework.http.MediaType;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.annotation.security.RolesAllowed;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static datawave.microservice.audit.replay.status.Status.ReplayState.CREATED;
+import static datawave.microservice.audit.replay.status.Status.ReplayState.IDLE;
+import static datawave.microservice.audit.replay.status.Status.ReplayState.RUNNING;
+import static datawave.microservice.audit.replay.status.Status.ReplayState.STOPPED;
+
+/**
+ * The ReplayController presents the REST endpoints for audit replay.
+ *
+ */
+@RestController
+@RequestMapping(path = "/v1/replay", produces = MediaType.APPLICATION_JSON_VALUE)
+@ConditionalOnProperty(name = "audit.replay.enabled", havingValue = "true")
```

Comment:
Is there a reason we don't want this enabled by default? (matchIfMissing = true)? Also, most things in spring boot that do a conditionalonproperty don't do havingProperty of true, since that would exclude "True" for example. The default is for the value to not equal "false", which works well for something that is enabled by default and the user override is to disable.

Reply:
This is mostly just a personal preference of mine to have non-critical features disabled unless they are explicitly enabled.  I hadn't thought about the 'true' vs. 'True' bit...  This is definitely not the only place I'm using ConditionalOnProperty this way.